File: linkedlist/prob1290.py
class Solution(object):
    def getDecimalValue(self, head):
        """
        :type head: ListNode
        :rtype: int
        """
        curr = head
        ans  = 0
        while curr:
            ans = (ans << 1) + curr.val
            curr = curr.next
        return ans
        

File: math/prob1290.kt
// Convert Binary Number in a Linked List to Integer
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun getDecimalValue(head: ListNode?): Int {

        var ans = 0
        var cur = head
        while (cur != null) {
            ans = ans * 2 + cur.`val`
            cur = cur.next
        }

        return ans

    }
}
